backend/test/dataset/test_list_dataset_files.py

```diff
@@ -43,103 +46,164 @@ def build_mock_event(dataset: DatasetModel):
 def test_list_dataset_files_success(mock_s3, mock_dataset_dao, mock_global_dataset):
     mock_dataset_dao.get.return_value = mock_global_dataset
     mock_s3.list_objects_v2.return_value = {
+        "Prefix": "global/datasets/example_dataset/",
```

Comment:
That's just how the `list_objects_v2` works. Since S3 doesn't actually have a concept of folders the `Prefix` parameter just limits results to objects starting with that prefix.